### ISSUE TITLE
testeth -- --filltests should not write the filled file when an error occurs

### ISSUE THREAD

**Issue by pirapira** [ethereum/cpp-ethereum#4464]
Currently, `testeth` writes the filled file to the filesystem even when the `expect` section is wrong in the filler.

For example, when I reproduce https://github.com/ethereum/tests/issues/305, the filled file is written to disk.

**Comment by pirapira** [comment]
@winsvega do you think the filled file should be written even when the filler is wrong?

**Comment by winsvega** [comment]
it should be written. actually I use filled file to see the post state and see what was actually wrong.

**Comment by pirapira** [comment]
@winsvega what about writing it with a wrong name.  Instead of writing `add.json`, what about writing `
add.failed.json`?

**Comment by winsvega** [comment]
hm. I actually did not had a need for this. But looks like for novice users it would be helpful

**Comment by pirapira** [comment]
OK. I changed the title and the description of this issue.

**Comment by winsvega** [comment]
so if the test filling went without error testeth should look for  testname.failed and remove it

**Comment by winsvega** [comment]
Wait. Actually if we are going to add a check that test folder does not contain any files that does not match test fillers. I think we not necessarily need this feature. 
If you are filling the tests, you just have to make sure that No errors occured. 
Otherwise you have debug information in the result .json file to investigate.

**Comment by pirapira** [comment]
Several times I ran testeth --filltests, saw errors, and committed the results accidentially.

**Comment by winsvega** [comment]
writing the file with another name would actually be inconvinient.

**Comment by pirapira** [comment]
@winsvega for what kind of use case?

**Comment by winsvega** [comment]
Development of a testfiller.

**Comment by pirapira** [comment]
People developing test fillers for EWASMTests got confused by the current behavior.  They could not figure out why the filled test does not pass testeth.

**Comment by winsvega** [comment]
you would never see this on filled tests unless you run --filltests command. 
we do not merge tests with debug information (because there would be an error on the travis run)

**Comment by pirapira** [comment]
Yes, they were developing their own tests.  I want more people to develop their own tests.

Problems happen before we merge, when they work on their own forks.

**Comment by winsvega** [comment]
@pirapira could you explain what is the problem and who has that problem and how?

cause I use the debug output to the same file when using --filltests
I do a test PR if --filltests shows no errors.

**Comment by pirapira** [comment]
@winsvega the problem is that the debug output can be easily added to a fork of `ethereum/tests` repository, and people who checked out the debug output will see failures that they don't understand.

**Comment by pirapira** [comment]
@winsvega the problem does not happen when you do a test PR.  The problem happens when other people do a test PR when you are not looking.

**Comment by winsvega** [comment]
so they are trying to PR a file even though testeth reported issues on the test...

**Comment by winsvega** [comment]
maybe this issue is about json schema check in the test repo?

**Comment by pirapira** [comment]
@winsvega no.  `testeth`'s error messages are ignored.  `testeth`'s status code is also ignored.  That's our userbase.  So `testeth` should not give useful filled tests to the users when some error happens.

**Comment by winsvega** [comment]
it is not really an error. but what if user adds his own wrong fields to the filled test? or testeth prints smth that we do not expect?
this is the issue for schema check.

**Comment by pirapira** [comment]
"What if user adds his own wrong fields"  This issue is not about that case.

"testeth prints smth that we do not expect" Who are "we" here?  The users?

"this is the issue for schema check"  No.  People should be able to use `testeth` without GitHub, Travis, or schema checkers.

**Comment by winsvega** [comment]
I am just saying that making new files would make things worse for me but wont solve the issue you initially talking about.

because if you add a test file with unexpected fields it will still be possible to merge without notice. 
even if you make a file with different name it will be possible to merge (in test repo) because no schema check is added.

**Comment by pirapira** [comment]
@winsvega then, it's easier to see what went wrong.
